jane/doc/extensions/_04-modes/intro.md

```diff
@@ -4,12 +4,198 @@ collectionName: Modes
 title: Intro
 ---
 
+<style>
+.table {
+    width: fit-content;
+    margin-left: auto;
+    margin-right: auto;
+    margin-bottom: 20px;
+    border-style: solid;
+    border-color: blue;
+    border-radius: 25px;
+    padding: 15px;
+    text-align: center;
+}
+</style>
+
 # Introduction to the Mode System
 
-The locality mode used for [Stack allocation](../stack/intro) has been
-extended to more axes, each tracking a property of values, in order to support
-features such as [Parallelism](../parallelism/intro) and
-[Uniqueness](../uniqueness/intro). This documentation gives a general
-introduction to the mode system.
+Modes are deep properties of values that are tracked by the OxCaml type
+system. Each mode is associated with a particular operation that may be
+performed on a value. The mode may be a *past* mode, which tracks whether the
+operation has happened to this value in the past, or a *future* mode, which
+tracks whether the operation is allowed to happen to this value in the future.
+
+Just like a value has a type, a value in OxCaml also has a mode. Types do *not*
+have a mode. That is, we do not have a type `string @ local` (say), but rather
+if we have `(x : string @ local)`, then `x` has type `string` and is at mode
+`local`. Modes also appear in the argument and return slots of a function type,
+so we can have `string @ local -> string option @ global` to describe a function
+whose argument will have the `local` mode and whose return will have the
+`global` mode. These modes appear in the function's type but are associated with the function's behavior, not the argument or result types (that is, there is still no `string @ local` or `string option @ global`).
+
+This page shows the modes that are currently supported. Each mode belongs to a
+modal *axis* determined by the operation it tracks and whether it is a past or
+future mode. The axes on this page are arranged with the least mode at the
+bottom and the greatest mode at the top.
+
+The type system supports *submoding*: values may move freely to greater modes
```

Comment:
```suggestion
The mode checking system supports *submoding*: values may move freely to greater modes
```

Reply:
Hmm.... I don't think we should break out "mode checker" from "type checker". Because of the modes on arrows, it would be impossible to implement without the other. (Well, maybe that's not true. It might be possible to do all type checking followed by mode checking, but it would be awkward in some cases, I think.) However, we should clarify that we use the term "type checker" inclusively to refer to checking both modes and types. (We could rename it the "type-and-mode-checker" or something, but I think that makes things worse.)